doc/user/content/serve-results/snowflake.md

```diff
@@ -0,0 +1,255 @@
+
+1. In your AWS account, find the IAM role you created in [Create an IAM role](#create-an-iam-role)
+   and, under **Trust relationships**, click **Edit trust policy**. Use the values
+   for the `STORAGE_AWS_IAM_USER_ARN` and `STORAGE_AWS_EXTERNAL_ID` properties
+   from the previous step to update the trust policy's `Principal` and
+   `ExternalId`, then click **Update policy**.
+
+## Step 3. Create a Snowflake external stage
+
+Back in Snowflake, create an [external stage](https://docs.snowflake.com/en/user-guide/data-load-S3-create-stage#external-stages) that uses the storage integration above and references your S3 bucket.
+
+```sql
+CREATE STAGE s3_stage
+  STORAGE_INTEGRATION = s3_int
+  URL = 's3://<bucket>/<prefix>/';
+```
+
+{{< note >}}
+To create a stage that uses a storage integration, the active user must have a
+role with the [`CREATE STAGE` privilege](https://docs.snowflake.com/en/sql-reference/sql/create-stage)
+for the active schema, as well as the [`USAGE` privilege](https://docs.snowflake.com/en/sql-reference/sql/grant-privilege#syntax)
+on the relevant storage integration.
+{{< /note >}}
+
+## Step 4. Import data into Snowflake
+
+To import the data stored in S3 into Snowflake, you can then create a table and
+use the [`COPY INTO`](https://docs.snowflake.com/en/sql-reference/sql/copy-into-table)
+command to load it from the external stage.
+
+{{< tabs >}}
+{{< tab "Parquet">}}
+
+Create a table with a single column of type [`VARIANT`](https://docs.snowflake.com/en/sql-reference/data-types-semistructured#variant):
+
+```sql
+CREATE TABLE s3_table_parquet (
+    col VARIANT
+);
+```
+
+Use `COPY INTO` to load the data into the table:
+
+```sql
+COPY INTO s3_table_parquet
+  FROM @S3_stage
+  FILE_FORMAT = (TYPE = 'PARQUET');
+```
+
+For more details on importing Parquet files staged in S3 into Snowflake, check the
+[Snowflake documentation](https://docs.snowflake.com/en/sql-reference/sql/copy-into-table#type-parquet).
+
+{{< /tab >}}
+
+{{< tab "CSV">}}
+
+Create a table with the same number of columns as the number of delimited
+columns in the input data file:
+
+```sql
+CREATE TABLE s3_table_csv (
+    col_1 INT,
+    col_2 TEXT,
+    col_3 TIMESTAMP
+);
+```
+
+Use `COPY INTO` to load the data into the table:
+
+```sql
+COPY INTO s3_table_csv
+  FROM @s3_stage
+  FILE_FORMAT = (TYPE = 'CSV');
+```
+
+For more details on importing CSV files staged in S3 into Snowflake, check the
+[Snowflake documentation](https://docs.snowflake.com/en/sql-reference/sql/copy-into-table#type-csv).
+
+{{< /tab >}}
+
+{{< /tabs >}}
+
+
+## Step 5. (Optional) Add scheduling
+
+Bulk exports to Amazon S3 using the `COPY TO` command are _one-shot_: every time
+you want to export results, you must run the command. To automate running bulk
+exports from Materialize to Snowflake on a regular basis, you can set up
+scheduling, for example using a simple `cron`-like service or an orchestration
+platform like Airflow or Dagster.
```

Comment:
@morsapaes did you look into using external tables here? https://docs.snowflake.com/en/sql-reference/sql/create-external-table#partitions-added-automatically-from-partition-column-expressions

I'm, like, 90% that's what we ought to be recommending here. It addresses @chuck-alt-delete's concern above about idempotency. It also gives a clean story around refreshes. You can offload a bunch of the complexity around refreshing with Snowflake's support for automatic refreshes when files on S3 are updated: https://docs.snowflake.com/en/user-guide/tables-external-s3

Reply:
I didn't, and saw no mention of it in their guides to integrate data from s3, but...I'll do some more toying around with external tables and also Snowpipe. Talk about having too many options, their documentation is pretty jarring.